akka-stream/src/main/scala/akka/stream/impl/Sources.scala

```diff
@@ -0,0 +1,131 @@
+/**
+ * Copyright (C) 2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.stream.impl
+
+import akka.stream.OverflowStrategies._
+import akka.stream._
+import akka.stream.stage._
+import scala.concurrent.{ Future, Promise }
+
+/**
+ * INTERNAL API
+ */
+private[akka] class QueueSource[T](maxBuffer: Int, overflowStrategy: OverflowStrategy) extends GraphStageWithMaterializedValue[SourceShape[T], SourceQueue[T]] {
+  trait SourceCompletion {
+    val completion = Promise[Unit]
+  }
```

Comment:
Build is showing warning as promise must be visible to external world (in SourceQueue instance)
```
[warn] /Users/alexandergolubev/Projects/akka/akka-stream/src/main/scala/akka/stream/impl/Sources.scala:121: reflective access of structural type member value completion should be enabled
[warn] by making the implicit value scala.language.reflectiveCalls visible.
[warn] This can be achieved by adding the import clause 'import scala.language.reflectiveCalls'
[warn] or by setting the compiler option -language:reflectiveCalls.
[warn] See the Scala docs for value scala.language.reflectiveCalls for a discussion
[warn] why the feature should be explicitly enabled.
[warn]       override def watchCompletion() = stageLogic.completion.future
```